### PR TITLE
feature/update theme and create Sign In UI

### PR DESCRIPTION
**Pull request by AlexisAllmnd**
### **This PR introduces the following key changes:**

**Theme Update:**
- Expanded and refined the color palette in Theme.kt to better support both light and dark modes, ensuring better contrast and consistency across the application.

- Customized Typography styles for different text elements, including headers, body text, labels, and buttons, improving readability and consistency.

- Applied custom shapes from Shapes.kt for buttons and other interactive elements.

**Sign In Screen Refactor:**

- Refactored the SignInScreen to dynamically use the updated MaterialTheme for colors, typography, and shapes.

- Improved the UI layout of the sign-in screen, ensuring a more cohesive design by aligning elements with the updated theme.

- Updated the Google Sign-In button to use the new theme settings, enhancing visual consistency.

These changes aim to improve the overall user experience by aligning all UI components with the updated design system, ensuring a smoother and more consistent look across the application.

### CODE REVIEW

File: app/src/main/java/com/github/se/project/screens/SignInScreen.kt [hunk 1]
```diff
@@ -1,29 +1,174 @@
...
+        Box(modifier = Modifier.fillMaxSize()) {
+          // Logo in top-left
+          Image(
+              painter = painterResource(id = R.drawable.logobrand), // Replace with your logo path
+              contentDescription = "Brand logo",
+              modifier =
+                  Modifier.size(200.dp) // Adjust size as needed
+                      .padding(16.dp) // Padding from top and left
+                      .align(Alignment.TopStart) // Align logo to top-left
+                      .testTag("logo"))
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/screens/SignInScreen.kt:53]: Logo is a great idea!

File: app/src/main/java/com/github/se/project/screens/SignInScreen.kt [hunk 2]
```diff
@@ -1,29 +1,174 @@
...
+                          Text(
+                              text =
+                                  when (page) {
+                                    0 ->
+                                        "Welcome to Pocket Tutor, Simplify learning and teaching with instant connections to the university community."
+                                    1 ->
+                                        "Get help or share your expertise – whether you're a student or a tutor, Pocket Tutor works for both."
+                                    else ->
+                                        "Pocket Tutor connects university students and tutors for quick, effective learning support."
+                                  },
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/screens/SignInScreen.kt:100]: Good slogans, adds to the user experience

File: app/src/main/java/com/github/se/project/screens/SignInScreen.kt [hunk 3]
```diff
@@ -1,29 +1,174 @@
...
+
+              // Terms and Conditions
+              Text(
+                  text =
+                      "By clicking 'Continue', you agree to the Terms of Service and Privacy Policy, and consent to the use of cookies related to PocketTutor.",
+                  style = Typography.bodySmall, // Use bodySmall style from the theme
+                  color = MaterialTheme.colorScheme.onBackground, // Use color from theme
+                  modifier = Modifier.padding(horizontal = 32.dp),
+                  textAlign = TextAlign.Center)
+            }
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/screens/SignInScreen.kt:157]: Do we have Terms and Conditions yet?

File: app/src/main/java/com/github/se/project/ui/theme/Color.kt [hunk 4]
```diff
@@ -2,10 +2,64 @@ package com.github.se.project.ui.theme
...
+val ErrorDark = Color(0xFFCF6679) // Light red for errors (adapted for dark mode)
+val SuccessDark = Color(0xFF81C784) // Light green for success
+
+// Disabled elements
+val DisabledDark = Color(0xFF424242) // Dark gray for disabled elements
+val DisabledOnBackgroundDark = Color(0xFF757575) // Disabled text on dark background
+
+// Accent and interaction
+val AccentDark = Color(0xFFFFD54F) // Light yellow accent for interactive elements
+val HoverDark = Color(0xFF2E2E2E) // Gray for hover state in dark mode
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/ui/theme/Color.kt:65]: Love the comments on where to use each colour, will be very helpful going forwards.

File: app/src/main/res/drawable/bg.png
Comment (A-M-00) [inline app/src/main/res/drawable/bg.png:1]: Adding a fancy background is a great idea!

File: app/src/main/res/drawable/logopocket.png
Comment (A-M-00) [inline app/src/main/res/drawable/logopocket.png:1]: Great logo! Gets the idea of what the app is from the first glance

File: app/src/main/java/com/github/se/project/ui/theme/Type.kt
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/ui/theme/Type.kt:1]: Good idea, having a consistent typography across different screens will help the app feel more harmonious

File: app/src/main/java/com/github/se/project/ui/theme/Shapes.kt
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/ui/theme/Shapes.kt:1]: Would be interesting to have other types of shapes defined, but this is a good initiative for now

**Review by A-M-00 — APPROVED** [review]
Overall, great work! I especially like the presets for coour, shapes, and typography, which will be a great help for any future screens we may write.

File: app/src/main/java/com/github/se/project/screens/SignInScreen.kt (hunk 3 above)
Comment (AlexisAllmnd) [reply app/src/main/java/com/github/se/project/screens/SignInScreen.kt:157]: no but imagine I imagine we will have some

**Review by A-M-00 — APPROVED** [review]
Important fix. Rest is still as good as it was for my last review

**Pull request merged**